### PR TITLE
made index as not required in schema level + using unique_column

### PR DESCRIPTION
**Pull request by ajaywk7**

### CODE REVIEW

File: elastalert/schema.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ definitions:
 
   filter: &filter {}
 
-required: [type, index, alert]
```
Comment (RashmiRam) [inline elastalert/schema.yaml:28]: why this change? It will be required for log alert rules?

**Review by RashmiRam — CHANGES_REQUESTED** [review]
Where is the default being set ?

File: elastalert/schema.yaml (hunk 1 above)
Comment (ajaywk7) [reply elastalert/schema.yaml:28]: its validating every rule with this schema first hand, if we dont remove this, we ll have to pass atleast a dummy index in each rule config.

Comment (ajaywk7) [reply elastalert/schema.yaml:28]: we will add a default index in config.yaml, which will be added to each rule when index not specified.

Comment (RashmiRam) [reply elastalert/schema.yaml:28]: In that case, why would this be required then?

Comment (ajaywk7) [reply elastalert/schema.yaml:28]: This is needed because, the flow goes like
-> Elastalert validates each rule with **schema.yaml** file ( so if we don't give index in rule - it ll throw error, thats why removed from schema)
-> Adds all the configs in **config.yaml** to all rules if its not present in individual rule config.

previously it was like, index field is a must in every indicidual rule config. as i removed index from required of schema.yaml, it can also be set in main config.yaml now.

**Review by RashmiRam — APPROVED** [review]
👍
